src/python/pants/base/exception_sink.py

```diff
@@ -216,13 +216,12 @@ def reset_log_location(cls, new_log_location: str) -> None:
     @classmethod
     def open_pid_specific_error_stream(cls, path):
         try:
-            ret = safe_open(path, mode="w+")
+            ret = safe_open(path, mode="w")
         except Exception:
             raise
 
         def unlink_if_empty():
-            ret.seek(0)
-            if ret.read(1) == "":
+            if os.path.getsize(path) == 0:
```

Comment:
These should probably be guarded to catch OSError and not re-raise if ENOENT in the same spirit. Don't even assume the file still exists.

Reply:
Having any sort of finalizer ever raise its own exception is a world of hurt; so good to make these pretty iron clad.

Reply:
Python seems to not mind: https://docs.python.org/3/library/atexit.html#atexit.register

On one hand maybe the user should be made aware that the file was removed and Pants wasn't expecting that. On the other hand... I actually can't think of the other hand.

Knowing that Python will gracefully-ish handle this, what do you think?

Reply:
Do the right thing and be robust & locally coherent without needing to go look at finer points of documentation. Even with those finer points, an harmless error gets spewed to a user.

Reply:
okie dokie